cmd/cstor-volume-mgmt/controller/volume-controller/handler.go

```diff
@@ -110,16 +139,27 @@ func (c *CStorVolumeController) cStorVolumeEventHandler(operation common.QueueOp
 		volStatus, err := volume.GetVolumeStatus(cStorVolumeGot)
 		if err != nil {
 			glog.Errorf("Error in getting volume status: %s", err.Error())
-			cStorVolumeGot.Status.Phase = apis.CStorVolumePhase(common.CVStatusError)
+			cStorVolumeGot.Status.Phase = apis.CStorVolumePhase(
+				common.CVStatusError,
+			)
 		} else {
 			cStorVolumeGot.Status.Phase = apis.CStorVolumePhase(volStatus.Status)
 			// if replicas are zero set the status as init
 			if len(volStatus.ReplicaStatuses) == 0 {
-				cStorVolumeGot.Status.Phase = apis.CStorVolumePhase(common.CVStatusInit)
+				cStorVolumeGot.Status.Phase = apis.CStorVolumePhase(
+					common.CVStatusInit,
+				)
 			}
 		}
+		cStorVolumeGot.Status.LastUpdateTime = metav1.Now()
+		if cStorVolumeGot.Status.Phase != lastKnownPhase {
```

Comment:
IMO we need to update `LastTransitionTime` even if there is change any one of the field under `status` section. In this PR we are updating only when there is phase change but `LastTransitionTime` is meant for status.
One way to do this by using `reflect.DeepEqual`

What would you say @vishnuitta @AmitKumarDas

Reply:
I think same thing need to do on `cStorPool and cvr`

Reply:
As we discussed `LastTransitionTime` is only for particular phase of the resource